### PR TITLE
Generate capi pacakge

### LINKED ISSUE #7181: The size of libmkldnn.so.0 is 0B in capi
Users encounter that lack of libmkldnn.so.0, and they `mv libmkldnn.so libmkldnn.so.0` currently to solve this problem. The reason is that libmkldnn.so.0 is a soft connection.
image
The download address of paddle.tgz is [here](https://guest@paddleci.ngrok.io/repository/download/Manylinux1_CpuAvxCp27cp27mu/.lastSuccessful/paddle.tgz).

### PR DESCRIPTION
**Pull request by Yancey1989**
related #7181

### CODE REVIEW

File: paddle/scripts/docker/build.sh [hunk 1]
```diff
@@ -193,13 +193,37 @@ EOF
...
 
+function gen_capi_package() {
+  if [[ ${WITH_C_API} == "ON" ]]; then
+    install_prefix="/paddle/build/capi_output"
+    rm -rf $install_prefix
+
+    make DESTDIR="$install_prefix" install
+
+    if [[ ${WITH_MKL:-OFF} == "ON" ]]; then
+      find ./third_party/install -name 'libmklml_gnu.so' -exec cp {} $install_prefix/usr/local/lib \;
```
Comment (luotao1) [inline paddle/scripts/docker/build.sh]: libmklml_gnu.so不用copy

File: paddle/scripts/docker/build.sh [hunk 2]
```diff
@@ -193,13 +193,37 @@ EOF
...
+      find ./third_party/install -name 'libmklml_intel.so' -exec cp {} $install_prefix/usr/local/lib \;
+      cp -P ./third_party/install/mkldnn/lib/* $install_prefix/usr/local/lib/
+    fi
```
Comment (luotao1) [inline paddle/scripts/docker/build.sh]: 照如下写法来代替204-206么？ @tensor-tang https://github.com/PaddlePaddle/Paddle/blob/5593858dd95a5921fbd938fcc8d7bb8003a2fdbf/python/setup.py.in#L89-L90

File: paddle/scripts/docker/build.sh [hunk 3]
```diff
@@ -193,13 +193,37 @@ EOF
...
+
+    find ./third_party/install -name 'libiomp5.so' -exec cp {} $install_prefix/usr/local/lib \;
```
Comment (luotao1) [inline paddle/scripts/docker/build.sh]: 209行的libiomp5只在`WITH_MKL=ON`的时候用，应该放到上面的if判断中。

**Comment by tensor-tang** [comment]
Could we make things simple, how about separate into 2 parts:
1. adding this package code in this PR
2. actually fix the issue #7181 next PR. I can follow this later.

**Comment by luotao1** [comment]
The ldd result is as follows now
image

File: paddle/scripts/docker/build.sh [hunk 4]
```diff
@@ -193,13 +193,37 @@ EOF
...
+  if [[ ${WITH_C_API} == "ON" ]]; then
+    install_prefix="/paddle/build/capi_output"
+    rm -rf $install_prefix
+
+    make DESTDIR="$install_prefix" install
+
+    if [[ ${WITH_MKL:-OFF} == "ON" ]]; then
+      find ./third_party/install -name 'libmklml_gnu.so' -exec cp {} $install_prefix/usr/local/lib \;
+      find ./third_party/install -name 'libmklml_intel.so' -exec cp {} $install_prefix/usr/local/lib \;
+      cp -P ./third_party/install/mkldnn/lib/* $install_prefix/usr/local/lib/
```
Comment (Xreki) [inline paddle/scripts/docker/build.sh]: 执行`make install`命令时，cmake中设定自动将一些third_party库拷贝到`CMAKE_INSTALL_PREFIX/third_party`目录下，目前会自动拷贝的库有`zlib`、`glog`、`gflags`、`protobuf`、`openblas`，如：

https://github.com/PaddlePaddle/Paddle/blob/d06bbb12e8657d31f7ed470100ebf5e7d8af3fbd/cmake/external/glog.cmake#L71-L78

可以考虑下要不要采用这种方式。

Comment (Yancey1989) [reply paddle/scripts/docker/build.sh]: I think it's a better way, and maybe @tensor-tang could consider with this.

**Pull request merged**